src/main/java/sdu/sem2/se17/persistence/data/UserHandlerImpl.java

```diff
@@ -15,26 +19,129 @@ public UserHandlerImpl(DataSource dataSource) {
+                PreparedStatement statement = connection.prepareStatement(""" 
+                    INSERT INTO User (username, email, password, usertype)
+                    VALUES (?, ?, ?, ?)
+                    RETURNING *
+                """)
+        ) {
+            configureStatement(user, statement);
+
+            try (ResultSet resultSet = statement.executeQuery()) {
+                if (resultSet.next()){
+                    result = Optional.of(map(resultSet));
+                }
+            }
+        } catch (SQLException throwables) {
+            throwables.printStackTrace();
+        }
+
+        return result;
     }
 
     @Override
     public Optional<User> read(long id) {
-        return Optional.empty();
+        Optional<User> result = Optional.empty();
+        try (
+                Connection connection = dataSource.getConnection();
+                PreparedStatement statement = connection.prepareStatement(""" 
+                    SELECT * FROM \"User\"
+                    WHERE id = ?
+                    RETURNING *
+                """)
+        ) {
+            statement.setLong(1, id);
+
+            try (ResultSet resultSet = statement.executeQuery()) {
+                if (resultSet.next()){
+                    result = Optional.of(map(resultSet));
+                }
+            }
+        } catch (SQLException throwables) {
+            throwables.printStackTrace();
+        }
+
+        return result;
     }
 
     @Override
     public void update(User user) {
-
+        try (
+                Connection connection = dataSource.getConnection();
+                PreparedStatement statement = connection.prepareStatement(""" 
+                    UPDATE User
+                    SET (name) = (username, password, email)
+                    (?)
+                    WHERE id = ?
+                """
+                )
+        ) {
+            configureStatement(user, statement);
+            statement.setLong(2, user.getId());
+            statement.execute();
+        } catch (SQLException throwables) {
+            throwables.printStackTrace();
+        }
     }
 
     @Override
     public void delete(long id) {
+        try (
+                Connection connection = dataSource.getConnection();
+                PreparedStatement statement = connection.prepareStatement("DELETE FROM User WHERE id = ?")
+        ) {
+            statement.setLong(1, id);
+            statement.execute();
+        } catch (SQLException throwables) {
+            throwables.printStackTrace();
+        }
 
     }
 
     @Override
     public Optional<User> findByUsername(String username) {
-        return Optional.empty();
+           Optional<User> result = Optional.empty();
+        try (
+                Connection connection = dataSource.getConnection();
+                PreparedStatement statement = connection.prepareStatement("""
+                    SELECT *
+                    FROM username
```

Comment:
Reference the User table and not the attribute.
Use there **where** statement to select what attribute you're searching for.